### ISSUE TITLE
[Bug]: With two segment_by columns, query time 20x after upgrade from 2.7.2 to 2.10.3 - only one index used

### ISSUE THREAD

**Issue by hanlaur** [timescale/timescaledb#5632]
### What type of bug is this?

Performance issue

### What subsystems and features are affected?

Query planner

### What happened?

After upgrading database from TimescaleDB 2.7 to 2.10, query times against compressed hypertable increased by 20x (from 0.5 seconds to 10seconds). When comparing execution plans, it appears that 2.7 used BitmapAnd over the two `segmentby` column related indices while 2.10 utilizes only one index.

TimescaleDB 2.7 did not yet have the composite index on `segmentby` columns, but in TimescaleDB 2.7 there are separate indices for the `segmentby` columns. If the table is created on 2.10 this problem is not occurring, because the composite index can be used. But old tables created in older version continue to use the old index structure after upgrade.

Script to reproduce the behavior is attached.

### TimescaleDB version affected

2.10.3

### PostgreSQL version used

13

### What operating system did you use?

Linux x64

### What installation method did you use?

Docker

### What platform did you run on?

On prem/Self-hosted

### How can we reproduce the bug?

```bash
-- docker run -d --name postgres -e POSTGRES_PASSWORD=abc --rm -it timescale/timescaledb:2.10.3-pg13
-- docker exec -it postgres psql -U postgres < timescale-2.10-one-index-only.sql

-- Start with old version of extension
DROP EXTENSION TIMESCALEDB;
\c
CREATE EXTENSION TIMESCALEDB VERSION "2.7.2";

-- Create schema
CREATE TABLE values (
  time       TIMESTAMPTZ NOT NULL,
  metric     INTEGER,
  device_id  TEXT NOT NULL,
  board_id    TEXT NOT NULL DEFAULT '',
  value      DOUBLE PRECISION NULL,
  UNIQUE (time, metric, device_id, board_id)
);
SELECT create_hypertable('values', 'time',
  chunk_time_interval => interval '1 day');
CREATE INDEX values_metric_device_time_idx ON values (metric, device_id, time DESC);

ALTER TABLE values SET (
  timescaledb.compress,
  timescaledb.compress_segmentby = 'metric, device_id',
  timescaledb.compress_orderby = 'board_id, time desc'
);

-- insert test data

do $$
declare
  t timestamptz;
begin
   for d in 0..0 loop
    for h in 0..23 loop
        t := timestamp '2021-07-01 00:00:00Z' + interval '1 day' * d + interval '1 hour' * h;
        raise notice 'adding data for %', t;
        for m in 0..9 loop
            insert into values (time,metric,device_id,board_id,value)
                select t, m, 'D'||s1, 'D'||s2, random() from generate_series(1,100) s1, generate_series(1,100) s2;
        end loop;
    end loop;
   end loop;
end; $$
;

-- compress and analyze
select compress_chunk(s) from show_chunks('values') s;
vacuum (full,analyze) values;

-- Execution with Timescale 2.7.2: BitmapAnd with both indices are used
EXPLAIN SELECT  device_id,board_id,sum(value) AS value
FROM values
WHERE time >= '2021-07-01 00:00:00Z'::TIMESTAMPTZ
AND time < '2021-07-02 00:00:00Z'::TIMESTAMPTZ
AND   device_id = ANY(ARRAY['D47'])
AND metric = 6
GROUP BY device_id, board_id;
--                                                                                 QUERY PLAN
-- --------------------------------------------------------------------------------------------------------------------------------------------------------------------------
--  HashAggregate  (cost=32.37..34.77 rows=240 width=14)
--    Group Key: _hyper_1_1_chunk.device_id, _hyper_1_1_chunk.board_id
--    ->  Custom Scan (DecompressChunk) on _hyper_1_1_chunk  (cost=3.29..9.87 rows=3000 width=14)
--          Filter: (("time" >= '2021-07-01 00:00:00+00'::timestamp with time zone) AND ("time" < '2021-07-02 00:00:00+00'::timestamp with time zone))
--          ->  Bitmap Heap Scan on compress_hyper_2_2_chunk  (cost=6.59..9.87 rows=3 width=134)
--                Recheck Cond: ((device_id = ANY ('{D47}'::text[])) AND (metric = 6))
--                Filter: ((_ts_meta_max_2 >= '2021-07-01 00:00:00+00'::timestamp with time zone) AND (_ts_meta_min_2 < '2021-07-02 00:00:00+00'::timestamp with time zone))
--                ->  BitmapAnd  (cost=6.59..6.59 rows=3 width=0)
--                      ->  Bitmap Index Scan on compress_hyper_2_2_chunk__compressed_hypertable_2_device_id__ts  (cost=0.00..1.61 rows=30 width=0)
--                            Index Cond: (device_id = ANY ('{D47}'::text[]))
--                      ->  Bitmap Index Scan on compress_hyper_2_2_chunk__compressed_hypertable_2_metric__ts_me  (cost=0.00..4.73 rows=300 width=0)
--                            Index Cond: (metric = 6)



-- Upgrade extension
\c
ALTER EXTENSION timescaledb UPDATE TO "2.10.3";
\c

-- Execution with Timescale 2.10.3: Only one of the indices is used. With real database this is leading to 20x query execution time: from 0.5sec to 10sec
EXPLAIN SELECT  device_id,board_id,sum(value) AS value
FROM values
WHERE time >= '2021-07-01 00:00:00Z'::TIMESTAMPTZ
AND time < '2021-07-02 00:00:00Z'::TIMESTAMPTZ
AND   device_id = ANY(ARRAY['D47'])
AND metric = 6
GROUP BY device_id, board_id;

-- -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
--  GroupAggregate  (cost=9.09..52.17 rows=240 width=14)
--    Group Key: _hyper_1_1_chunk.device_id, _hyper_1_1_chunk.board_id
--    ->  Custom Scan (DecompressChunk) on _hyper_1_1_chunk  (cost=9.09..27.27 rows=3000 width=14)
--          Filter: (("time" >= '2021-07-01 00:00:00+00'::timestamp with time zone) AND ("time" < '2021-07-02 00:00:00+00'::timestamp with time zone))
--          ->  Index Scan using compress_hyper_2_2_chunk__compressed_hypertable_2_device_id__ts on compress_hyper_2_2_chunk  (cost=0.28..27.27 rows=3 width=134)
--                Index Cond: (device_id = ANY ('{D47}'::text[]))
--                Filter: ((_ts_meta_max_2 >= '2021-07-01 00:00:00+00'::timestamp with time zone) AND (_ts_meta_min_2 < '2021-07-02 00:00:00+00'::timestamp with time zone) AND (metric = 6))
-- (7 rows)
```
```

**Comment by hanlaur** [comment]
> Hello @hanlaur thank you very much for bringing this to our attention. As a possible workaround, have you tried to create the composite index so that it gets picked by the planner instead?

Thanks @konskov . I definitely could try it. As these indices are automatically managed by TimescaleDB extension, I was not sure what is the correct procedure to create the index and drop the two old indices so that they would be automatically applied by TimescaleDB for new future chunks as well. I would not like to break TimescaleDB internal metadata and future upgrades :)

Would you have a tip on correct "migration" procedure from old format to new format? Do I just do the following and that is it?
```sql
drop index _timescaledb_internal._compressed_hypertable_2_device_id__ts_meta_sequence_num_idx ;
drop index _timescaledb_internal._compressed_hypertable_2_metric__ts_meta_sequence_num_idx;
create index on _timescaledb_internal._compressed_hypertable_2 (metric, device_id, _ts_meta_sequence_num);
```

**Comment by konskov** [comment]
hello @hanlaur - yes, these are exactly the steps you need to follow.

**Comment by kgyrtkirk** [comment]
notes:
* 2.7.3 create the table with 2 indexes:
```
    "compress_hyper_2_2_chunk__compressed_hypertable_2_device_id__ts" btree (device_id, _ts_meta_sequence_num)
    "compress_hyper_2_2_chunk__compressed_hypertable_2_metric__ts_me" btree (metric, _ts_meta_sequence_num)
```
* 2.10.3 creates the same table with a single index
```
    "compress_hyper_2_2_chunk__compressed_hypertable_2_metric_device" btree (metric, device_id, _ts_meta_sequence_num)
```
* this second index fits even more with the query 
* the provided test case runs under kinda the same amount of time regrardless how indexes are used

**Comment by kgyrtkirk** [comment]
index creation was altered in #4314; first released with 2.8.0
